nemo_skills/code_execution/sandbox.py

```diff
@@ -262,11 +275,21 @@ def is_output_correct(self, pred_output, gt_output, include_percentage=True, tol
 sys.stdout = stdout
 print(json.dumps({{"result": output, "error_message": error_message}}))
 """
-        request = self._prepare_request(TO_EXECUTE, timeout)
+        elif language == "lean4":
+            TO_EXECUTE = pred_output
+
+        request = self._prepare_request(TO_EXECUTE, timeout, language)
         try:
             output = self._send_request(request, timeout)
-        except requests.exceptions.Timeout:
-            output = {'result': False, 'error_message': 'timeout'}
+        except Exception as e:
+            output = {'result': False, 'error_message': e}
+
+        if language == "lean4" and "process_status" in output:
```

Comment:
```suggestion
        if language == "lean4"
```
Why the and part is needed?